base_layer/core/src/proof_of_work/monero_rx/helpers.rs

```diff
@@ -75,11 +75,17 @@ fn get_random_x_difficulty(input: &[u8], vm: &RandomXVMInstance) -> Result<(Diff
 pub fn verify_header(header: &BlockHeader) -> Result<MoneroPowData, MergeMineError> {
     let monero_data = MoneroPowData::from_header(header)?;
     let expected_merge_mining_hash = header.merge_mining_hash();
-    let extra_field = ExtraField::try_parse(&monero_data.coinbase_tx.prefix.extra)
-        .map_err(|_| MergeMineError::DeserializeError("Invalid extra field".to_string()))?;
+
+    // Parsing an extra field from bytes will always return a valid extra field, even if it does not represent the
+    // original extra field. As per Monero consensus rules, an error here will not represent a failure to deserialize a
+    // block, so no need to error here.
+    let extra_field = match ExtraField::try_parse(&monero_data.coinbase_tx.prefix.extra) {
+        Ok(val) => val,
+        Err(val) => val,
```

Comment:
I know this is the Monero way of handling things, but we should at least include a log here, so we don't just suppress it.

Reply:
It's such an antipattern. Why use `try_parse` if it always returns with a value. It should just be `parse`.

Reply:
They do this for clients interested in the data irrespective. They also have a wrapper on top of `ExtraField::try_parse`, `RawExtraField::try_parse`, that will always return an `ExtraField`.